core/services/encryption/EncryptionKeyManager.php

```diff
@@ -0,0 +1,272 @@
+     * @return string
+     * @throws Exception
+     * @throws OutOfBoundsException
+     */
+    public function getEncryptionKey($encryption_key_identifier = '', $generate = true)
+    {
+        $encryption_key_identifier = $encryption_key_identifier ?: $this->default_encryption_key_id;
+        // ensure keys are loaded
+        $this->retrieveEncryptionKeys();
+        // if encryption key has not been set
+        if (! isset($this->encryption_keys[ $encryption_key_identifier ])) {
+            if ($generate) {
+                $this->addEncryptionKey($encryption_key_identifier);
+            } else {
+                throw new OutOfBoundsException(
+                    sprintf(
+                        esc_html__('The "%1$s" encryption key was not found or is invalid.', 'event_espresso'),
+                        $encryption_key_identifier
+                    )
+                );
+            }
+        }
+        return $this->encryption_keys[ $encryption_key_identifier ];
+    }
+
+
+    /**
+     * creates a new encryption key
+     *
+     * @param bool $strong if true (default) will attempt to generate a cryptographically secure key
+     * @return string
+     * @throws Exception
+     */
+    public function generateEncryptionKey($strong = true)
+    {
+        return $strong && PHP_VERSION_ID >= 70100
+            ? $this->generateStrongEncryptionKey()
+            : $this->generateWeakEncryptionKey();
+    }
+
+
+    /**
+     * creates a new cryptographically secure encryption key
+     *
+     * @return string
+     * @throws Exception
+     */
+    protected function generateStrongEncryptionKey()
+    {
+        // bit_depth needs to be divided by 8 to convert to bytes
+        return $this->base64_encoder->encodeString(random_bytes($this->bit_depth / 8));
+    }
+
+
+    /**
+     * creates a new encryption key that should not be trusted to be cryptographically secure
+     *
+     * @return string
+     * @throws Exception
+     */
+    protected function generateWeakEncryptionKey()
+    {
+        // @see http://stackoverflow.com/questions/637278/what-is-the-best-way-to-generate-a-random-key-within-php
+        $iterations    = ceil($this->key_length / 40);
+        $random_string = '';
+        for ($i = 0; $i < $iterations; $i++) {
+            $random_string .= sha1(microtime(true) . mt_rand(10000, 90000));
+        }
+        $random_string = (string) substr($random_string, 0, $this->key_length);
+        return $this->base64_encoder->encodeString($random_string);
+
+    }
+
+
+    /**
+     * @return int
+     */
+    public function bitDepth()
+    {
+        return $this->bit_depth;
+    }
+
+
+    /**
+     * @param int $bit_depth options are 64, 128, 192, or 256
+     */
+    public function setBitDepth($bit_depth)
+    {
+        $bit_depth       = absint($bit_depth);
+        $this->bit_depth = in_array($bit_depth, $this->bit_depth_options) ? $bit_depth : 128;
```

Comment:
Why not strict comparison?
```suggestion
        $this->bit_depth = in_array($bit_depth, $this->bit_depth_options, true) ? $bit_depth : 128;
```

Reply:
oh good catch, thnx, i totally missed that, will fix